### ISSUE TITLE
How do you add an Azure ChatGPT deployment as the OPENAI_API_BASE?

### ISSUE THREAD

**Issue by dnhkng** [OpenBMB/ChatDev#191]
Hi,

I have access to ChatGPT via Azure, and I can access it with curl:

```
curl "https://my_deployment.openai.azure.com/openai/deployments/my_gpt35turbo/completions?api-version=2023-05-15"\
  -H "Content-Type: application/json" \
  -H "api-key: my_secret_key" \
  -d "{
  \"prompt\": \"Once upon a time\",
  \"max_tokens\": 50
}"   
```

How do I use this with Chatdev? I've tried multiple ways, like this:
```
OPENAI_API_BASE="https://my_deployment.openai.azure.com/openai/deployments/my_gpt35turbo?api-version=2023-05-15" OPENAI_API_KEY=my_secret_key python3 run.py --task "Snake game in pure html" --name "WebSnake"
```

But I always see this in the logs:
`[2023-17-10 09:03:10 INFO] error_code=404 error_message='Resource not found' error_param=None error_type=None message='OpenAI API error received' stream_error=False
`

Any tips on how to correctly set the OPENAI_API_BASE for Azure users? Does the deployment name have to match? ie it must be set as 'GPT_3_5_TURBO' on Azure? Or can I modify that in the ChatDev code?

Thanks!

**Comment by dnhkng** [comment]
UPDATE: OK, I have gotten a bit further:

```
OPENAI_API_BASE="https://my_deployment.openai.azure.com/" OPENAI_API_TYPE=azure OPENAI_API_MODEL=my_model OPENAI_API_VERSION=2023-05-15 OPENAI_API_KEY= my_secret_key python3 run.py --task "Snake game in pure html" --name "WebSnake"
```

This solves the 'Resource not found' error, but now generates this error:

```
Must provide an 'engine' or 'deployment_id' parameter to create a <class 'openai.api_resources.chat_completion.ChatCompletion'>
```

Im not sure what ''engine' or 'deployment_id' should be but adding:
`OPENAI_API_ENGINE=my_model`
to the command does not help.

**Comment by dnhkng** [comment]
OK, fixed in pull #192
